os/fs/smartfs/smartfs_smart.c

```diff
@@ -736,6 +747,20 @@ static ssize_t smartfs_write(FAR struct file *filep, const char *buffer, size_t
 		}
 	}
 
+#ifdef CONFIG_SMARTFS_USE_SECTOR_BUFFER
+	/* If there is unsynced data in the sf->buffer and the control is now leaving the
+	 * scope of this function, the correct header should also be present in sf->buffer
+	 * for smartfs_append_data or smartfs_sync_internal. Hence we read the header now.
+	 */
+	if (sf->bflags & SMARTFS_BFLAG_DIRTY && byteswritten > 0) {
```

Comment:
Instead of this option, I think we can use sf->curroffset simply

Reply:
You mean
` if (sf->curroffset > size)`
?

Reply:
yes